### ISSUE TITLE
[Repo Rearchitecture] Determine the future of schemas and docs

### ISSUE THREAD

**Issue by rudouglas** [newrelic/newrelic-quickstarts#629]
### Description

We currently have a [directory of schemas](https://github.com/newrelic/newrelic-quickstarts/tree/main/utils/schemas) and a job in the release workflow that generates schema docs using  them:

 - [Generate-schema-docs](https://github.com/newrelic/newrelic-quickstarts/blob/e4ae0a4b5a10fb2da9de324ffdfae8cbbcd206c7/.github/workflows/release.yml#L101) 
         - Script: [utils/schema-doc-generator/script.py](https://github.com/newrelic/newrelic-quickstarts/tree/main/utils/schema-doc-generator)

Because the validation will now be done via NerdGraph and that will be the single source of truth for quickstart schema, we will need to determine if we need these any more. The directory should probably be removed, but it may still be useful to provide schema docs publicly but we will need to pull the schema information from NerdGraph in order to do this

### Acceptance Criteria

 - [x] Determine how much effort it would be to update the [docs](https://github.com/newrelic/newrelic-quickstarts/tree/main/docs) based on the schemas.
 - [x] Discuss the value in keeping public schema docs up to date and the work involved. Determine if we have page view data on the markdown files found in the [docs](https://github.com/newrelic/newrelic-quickstarts/tree/main/docs) directory. 
 - [x] Create a follow up ticket for the work to make this change.

**Comment by aswanson-nr** [comment]
I looked into a few code generators:
* [SpectaQL - generate static docs for a GraphQL schema | GraphQL Blog](https://blog.graphqleditor.com/spectaql)
* [The Rover CLI](https://www.apollographql.com/docs/rover/)
* [GraphQL Code Generator](https://www.graphql-code-generator.com/)

None seem to be an "out of the box" solution that can use introspection queries, most rely on direct access to the schema.

I also put together an introspection query myself, it's not complete but it gives you an idea of what we can do:
```gql
query IntrospectionQuery {
  __type(name: "Nr1CatalogQuickstart"){
    types {
      ...FullType
    }
    directives {
      name
      locations
      args {
        ...InputValue
      }
    }
  }
}
fragment FullType on __Type {
  kind
  name
  fields(includeDeprecated: true) {
    name
    args {
      ...InputValue
    }
    type {
      ..TypeRef
    }
    isDeprecated
    deprecationReason
  }
  inputFields {
    ...InputValue
  }
  interfaces {
    ...TypeRef
  }
  enumValues(includeDeprecated: true) {
    name
    isDeprecated
    deprecationReason
  }
  possibleTypes {
    ...TypeRef
  }
}
fragment InputValue on __InputValue {
  name
  type { ...TypeRef }
  defaultValue
}
fragment TypeRef on __Type {
  kind
  name
  ofType {
    kind
    name
    ofType {
      kind
      name
      ofType {
        kind
        name
        ofType {
          kind
          name
          ofType {
            kind
            name
            ofType {
              kind
              name
              ofType {
                kind
                name
              }
            }
          }
        }
      }
    }
  }
}
```
There is always the option of manually parsing the return value and producing markdown files

**Comment by zstix** [comment]
It looks like the part of the nerdgraph schema we care about can be found in our internal nerd-graph repo (hint: `nr1_catalog/introspection.json`). We should ask the fine folks on the unified team if theres a way that we can ping their service for this information / file. We could probably set up a Github action to grab this information and regenerate the docs every so often.

**Comment by zstix** [comment]
_If_ we need to query GraphQL for information about the schema, here's an example of how we would do that for `Nr1CatalogQuickstartMetadataAlertConditionInput`:

```graphql
query ($typeName: String!) {
  __type(name: $typeName) {
    ...TypeInfo
    inputFields {
      ...InputTypeInfo
      type {
        ...TypeInfo
        ofType {
          ...TypeInfo
          enumValues {
            name
          }
        }
      }
    }
  }
}

fragment TypeInfo on __Type {
  name
  kind
  description
}

fragment InputTypeInfo on __InputValue {
  name
  description
}
```

Or something like Alec's query / the query in our internal repository.

TL;DR: we could write a query to get all the information we need, but it would be easiest to just get the `introspection.json` files we care about.

**Comment by jerelmiller** [comment]
@zstix do you need this to be automated? You could always just ping our service directly for the schema. You'll have to be behind the VPN to do so, but if you're ok manually running it from time-to-time, that could be a way to go. I'd avoid commiting the internal URI anywhere, but you could always have it run off of an environment variable:

```sh
URL=https://that.one.internal.service/graphql node regenerate-docs.js
```

**Comment by rudouglas** [comment]
I've actually just done the work in this ticket because it ended up being super simple
